circus-train-s3-s3-copier/src/main/java/com/hotels/bdp/circustrain/s3s3copier/aws/JceksAmazonS3ClientFactory.java

```diff
@@ -103,13 +103,27 @@ private String regionForUri(AmazonS3 client, AmazonS3URI uri) {
     return bucketRegion;
   }
 
+  private AmazonS3ClientBuilder applyClientConfigurations(AmazonS3ClientBuilder builder, S3S3CopierOptions s3s3CopierOptions) {
+    ClientConfiguration clientConfiguration = new ClientConfiguration();
+
+    if (s3s3CopierOptions.getMaxThreadPoolSize() != -1) {
+      clientConfiguration.withMaxConnections(s3s3CopierOptions.getMaxThreadPoolSize());
+    }
+
+    return builder.withClientConfiguration(clientConfiguration);
```

Comment:
Why do we need this here as well as in the TransferManagerFactory?

Reply:
The call in TransferManagerFactory sets the max threads that are available for its pool but doesn't update the default connection limit that the S3Client has of 50. So if you were to set a thread pool of 100, you'd start hitting connection failures.

From my research into it, the most efficient scenario is to match # of threads to # of available connections.

Reply:
OK